activerecord/lib/active_record/associations/set_through_association_source_target.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Associations
+    module SetThroughAssociationSourceTarget
+      def set_through_association_source_target(record)
+        through_reflection = self.reflection.active_record.reflections.values.find { |r|
+          r.through_reflection && r.through_reflection.name == self.reflection.name
+        }
```

Comment:
What are the keys for `self.reflection.active_record.reflections`?  Is it not the name?

Reply:
Yes, it's the name, but it's the name of the reflection itself, whereas I need a through reflection's name. So for example given a comment model:

```ruby
class Comment < ActiveRecord::Base
  belongs_to :post
  has_one :post_author, through: :post
end
```

I'm running this method on `comment.post = post` and I'm looking for a `post_author` reflection. So I'm looking for a reflection where `reflection.through_reflection.name == :post`.